projects/ad7746-ebz/src/app/headless.c

```diff
@@ -0,0 +1,170 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/platform.h>
+#include "adi_initialize.h"
+#include <drivers/pwr/adi_pwr.h>
+
+#include "ad7746.h"
+#include "i2c.h"
+#include "delay.h"
+#include "print_log.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "uart_stdio.h"
+
+
+int32_t platform_init()
```

Comment:
Can you use: https://github.com/analogdevicesinc/no-OS/blob/1695b4d602ad39bb6ebff5c43b1724c138a5f590/drivers/platform/aducm3029/platform_init.h#L45
Is the same function.

Reply:
ok, didn't know there is a default one, will use it, although to be honest, having it under drivers/platform/aducm3029 makes no sense at all because this is application level stuff - only in application you want to care what speed are you running on and set those prescalers to 1 - you can't have a file in drivers/platform/aducm3029 that sets the same settings for all possible applications...

Reply:
Maybe an init param can be added, but as example project which most project will be, I think is good enough.

Reply:
This mostly works because you only care about ADuCM3029, but what about if you cared about all ADI microcontrollers ? 
https://www.analog.com/en/products/processors-microcontrollers/microcontrollers/precision-microcontrollers.html

A platform should handle all of them, not just 1. Our current aducm supports just 1, so that platform_init() is ok for now.